raiden/transfer/channel.py

```diff
@@ -337,6 +349,30 @@ def is_valid_amount(
 
 
 def is_valid_signature(
+    data: bytes, signature: Signature, sender_address: Address
+) -> SuccessOrError:
+    try:
+        signer_address = recover(data=data, signature=signature)
+        # InvalidSignature is raised by raiden.utils.signer.recover if signature
+        # is not bytes or has the incorrect length
+        #
+        # ValueError is raised if the PublicKey instantiation failed, let it
+        # propagate because it's a memory pressure problem.
+        #
+        # Exception is raised if the public key recovery failed.
+    except Exception:  # pylint: disable=broad-except
```

Comment:
We do not raise exceptions in the state machines. In this case, validating the signature against the data failed so that the caller does not act upon the received state change.

Reply:
good reason! then perhaps the comment is outdated and should be removed/reworded?

Reply:
Will do

Reply:
I removed the part which mentions exception propagation